src/effects/effectrack.cpp

```diff
@@ -242,23 +242,34 @@ PerGroupRack::PerGroupRack(EffectsManager* pEffectsManager,
         : EffectRack(pEffectsManager, pChainManager, iRackNumber, group) {
 }
 
-EffectChainSlotPointer PerGroupRack::addEffectChainSlotForGroup(const QString& group) {
-    if (m_groupToChainSlot.contains(group)) {
+EffectChainSlotPointer PerGroupRack::addEffectChainSlotForGroup(const QString& groupName) {
+    if (m_groupToChainSlot.contains(groupName)) {
         qWarning() << "PerGroupRack" << getGroup()
-                   << "group is already registered" << group;
-        return getGroupEffectChainSlot(group);
+                   << "group is already registered" << groupName;
+        return getGroupEffectChainSlot(groupName);
     }
 
     int iChainSlotNumber = m_groupToChainSlot.size();
     QString chainSlotGroup = formatEffectChainSlotGroupForGroup(
-        getRackNumber(), iChainSlotNumber, group);
+        getRackNumber(), iChainSlotNumber, groupName);
     EffectChainSlot* pChainSlot = new EffectChainSlot(this, chainSlotGroup,
                                                       iChainSlotNumber);
     EffectChainSlotPointer pChainSlotPointer(pChainSlot);
     addEffectChainSlotInternal(pChainSlotPointer);
-    m_groupToChainSlot[group] = pChainSlotPointer;
+    m_groupToChainSlot[groupName] = pChainSlotPointer;
+
+    // TODO(rryan): remove.
```

Comment:
Yea... eh. Hm. This is annoying. In this part of the code we only have access to a list of registered channels so we have to linear search through it to get the handle that matches the channel group string. I'm not actually sure how to make it better right now other than giving DlgPrefEq access to EngineMaster in order to lookup the handle from the string in its handle factory. 

Since these methods are only called on preference changes and they run in the main thread I don't think a linear search through ~20 items is that bad. Maybe remove the TODOs?